easybuild/easyblocks/generic/rpackage.py

```diff
@@ -215,16 +205,15 @@ def run(self):
             lib_install_prefix = os.path.join(self.installdir, self.cfg['exts_subdir'])
             mkdir(lib_install_prefix, parents=True)
 
-        if self.patches:
-            super(RPackage, self).run(unpack_src=True)
-        else:
-            super(RPackage, self).run()
-
         if self.src:
+            super(RPackage, self).run(unpack_src=True)
```

Comment:
This is also significantly changing the logic, why don't we need to check for patches anymore?

Reply:
Previous: Extract only if there are patches   
Now: Extract always because config.guess needs to be "patched" (so there are "patches")

Placed inside `self.src` check because the super-`run` was a no-op anyway if neither patches nor src were set

Reply:
OK, thanks for clarifying